pybossa/view/account.py

```diff
@@ -1151,6 +1151,61 @@ def add_metadata(name):
     flash("Input saved successfully", "info")
     return redirect(url_for('account.profile', name=name))
 
+
+@blueprint.route('/<user_name>/taskbrowse_bookmarks/<short_name>', methods=['GET', 'POST'])
+@login_required
+def taskbrowse_bookmarks(user_name, short_name):
+    user = user_repo.get_by_name(name=user_name)
+    if user is None:
+        return abort(404)
+    if current_user.name != user_name:
+        return abort(403)
+
+    all_bookmarks = user.info.get('taskbrowse_bookmarks', {})
+    proj_bookmarks = all_bookmarks.get(short_name, {})
```

Comment:
Are you considering future options where we could store bookmarks other than project bookmarks or this is going to store different bookmarks, and all of the are project taskbrowse bookmarks? If its later, can `all_bookmarks` be termed as either `bookmarks` or `taskbrowse_bookmarks`?

Reply:
As the activity of creating/updating bookmarks is less frequent, consider caching the information fetched from db for certain time. Cache can be cleared upon each update, so that latest information is fetched after update. This is to avoid db calls if someone makes frequent calls to this endpoint.

Reply:
refactor to `taskbrowse_bookmarks` via https://github.com/bloomberg/pybossa/pull/832/commits/36a4707fe170566458061b887e5ec487b7952f31
caching via https://github.com/bloomberg/pybossa/pull/832/commits/d4bf637774cf0171d98830d01ad116349e0f0cce